src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -2117,71 +2117,18 @@ public static bool IsLabelContext(this SyntaxTree syntaxTree, int position, Canc
         public static bool IsNameOfContext(this SyntaxTree syntaxTree, int position, SemanticModel semanticModelOpt = null, CancellationToken cancellationToken = default(CancellationToken))
         {
             var token = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
-            token = token.GetPreviousTokenIfTouchingWord(position);
 
-            // nameof(Foo.|
-            // nameof(Foo.Bar.|
-            // Locate the open paren.
-            if (token.IsKind(SyntaxKind.DotToken))
-            {
-                // Could have been parsed as member access
-                if (token.Parent.IsKind(SyntaxKind.SimpleMemberAccessExpression))
-                {
-                    var parentMemberAccess = token.Parent;
-                    while (parentMemberAccess.IsParentKind(SyntaxKind.SimpleMemberAccessExpression))
-                    {
-                        parentMemberAccess = parentMemberAccess.Parent;
-                    }
-
-                    if (parentMemberAccess.IsParentKind(SyntaxKind.Argument) &&
-                        parentMemberAccess.Parent.IsChildNode<ArgumentListSyntax>(a => a.Arguments.FirstOrDefault()))
-                    {
-                        token = ((ArgumentListSyntax)parentMemberAccess.Parent.Parent).OpenParenToken;
-                    }
-                }
+            
 
-                // Could have been parsed as a qualified name.
-                if (token.Parent.IsKind(SyntaxKind.QualifiedName))
-                {
-                    var parentQualifiedName = token.Parent;
-                    while (parentQualifiedName.IsParentKind(SyntaxKind.QualifiedName))
-                    {
-                        parentQualifiedName = parentQualifiedName.Parent;
-                    }
+            var invocationExpression = token.GetAncestor(a => a.IsKind(SyntaxKind.InvocationExpression)) as InvocationExpressionSyntax;
```

Comment:
I really didn't think this would matter for nameof. You can't even legally put expressions like that in a nameof, so if that code exists and nameof doesn't bind, you're either in broken code because you haven't created a nameof method yet or you're in broken code because you haven't realized that your subexpression is illegal within the nameof.